tests/unit/net/fetch.rs

```diff
@@ -60,6 +61,18 @@ fn test_fetch_response_is_not_network_error() {
 }
 
 #[test]
+fn test_fetch_on_bad_port_is_network_error() {
```

Comment:
Any chance we can write this down also as a web platform test (if it doesn't exist)?